rule-audit-client/src/utils/prosemirror.ts

```diff
@@ -0,0 +1,20 @@
+import { EditorState } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+import { Schema, DOMParser } from "prosemirror-model";
+import { marks, schema } from "prosemirror-schema-basic";
+import { addListNodes } from "prosemirror-schema-list";
+import { getBlocksFromDocument } from "@guardian/prosemirror-typerighter";
+
+const appSchema = new Schema({
+  nodes: addListNodes(schema.spec.nodes as any, "paragraph block*", "block"),
+  marks
+});
+
+const parser = DOMParser.fromSchema(appSchema);
+
+export const getBlocksFromHtmlString = (htmlStr: string) => {
+  const element = document.createElement("div");
```

Comment:
I don't think it matters here since we trust the source of our content but I've used [DocumentFragment](https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment) in the past to parse HTML

Reply:
This is absolutely what should be happening here, good catch 👍

Reply:
Hm, on further consideration I don't think `DocumentFragment` helps us here – you must still append a valid Node to the created fragment. As you've approved, I'll merge for now, although happy to discuss and amend if I've misunderstood!

On my travels to understand this better, though, I did discover DOMParser, which has an advantage in that it doesn't trigger browser fetches on parse for elements with src attributes, so I've used that instead.